cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -274,32 +270,31 @@ func instanceIdsToNamesMap(instanceProviderIds []string) map[string]bool {
 	return instanceNames
 }
 
-func (client *autoscalingGceClientV1) waitForOp(operation *gce.Operation, project, zone string, isDeletion bool) error {
-	pollInterval := client.operationPollInterval
-	if isDeletion {
-		pollInterval = client.operationDeletionPollInterval
-	}
-	for start := time.Now(); time.Since(start) < client.operationWaitTimeout; time.Sleep(pollInterval) {
-		klog.V(4).Infof("Waiting for operation %s %s %s", project, zone, operation.Name)
-		registerRequest("zone_operations", "get")
-		if op, err := client.gceService.ZoneOperations.Get(project, zone, operation.Name).Do(); err == nil {
-			klog.V(4).Infof("Operation %s %s %s status: %s", project, zone, operation.Name, op.Status)
-			if op.Status == "DONE" {
-				if op.Error != nil {
-					errBytes, err := op.Error.MarshalJSON()
-					if err != nil {
-						errBytes = []byte(fmt.Sprintf("operation failed, but error couldn't be recovered: %v", err))
-					}
-					return fmt.Errorf("error while getting operation %s on %s: %s", operation.Name, operation.TargetLink, errBytes)
-				}
+func (client *autoscalingGceClientV1) WaitForOperation(operationName, operationType, project, zone string) error {
+	ctx, cancel := context.WithTimeout(context.TODO(), client.operationWaitTimeout)
+	defer cancel()
+
+	for {
```

Comment:
Reading the documentation, it seems that if the server is overloaded, the Wait call can return immediately. I'd still keep some interval between retries for this scenario, but IMO it doesn't make sense to differentiate the timeouts between different operations anymore.

Reply:
Fully agree with that, thanks for mentioning! Yeah, I don't think that it makes sense to make it configurable even, but the tests may struggle from that potentially. 

The only thing that bothers me is that there's no mention in docs how to identify that server is overloaded based on response (so that sleep would be called only in this scenario)

Reply:
As a result - added `operationPollInterval`, but made it not configurable from outside of the package, so that only tests may access that for modification